logstreams/src/test/java/io/camunda/zeebe/logstreams/util/TestEntry.java

```diff
@@ -0,0 +1,97 @@
+import io.camunda.zeebe.protocol.record.intent.Intent;
+import io.camunda.zeebe.util.ReflectUtil;
+import org.assertj.core.api.AbstractObjectAssert;
+import org.assertj.core.api.Assertions;
+
+public record TestEntry(
+    long key, int sourceIndex, UnifiedRecordValue recordValue, RecordMetadata recordMetadata)
+    implements LogAppendEntry {
+
+  public static LogAppendEntry ofDefaults() {
+    return new TestLogAppendEntryBuilder().build();
+  }
+
+  public static TestLogAppendEntryBuilder builder() {
+    return new TestLogAppendEntryBuilder();
+  }
+
+  public static LogAppendEntry ofKey(final long key) {
+    return new TestLogAppendEntryBuilder().withKey(key).build();
+  }
+
+  public static final class TestEntryAssert
+      extends AbstractObjectAssert<TestEntryAssert, LogAppendEntry> {
+
+    private TestEntryAssert(final LogAppendEntry testEntry, final Class<?> selfType) {
+      super(testEntry, selfType);
+    }
+
+    public static TestEntryAssert assertThatEntry(final LogAppendEntry entry) {
+      return new TestEntryAssert(entry, TestEntryAssert.class);
+    }
+
+    public TestEntryAssert matchesLoggedEvent(final LoggedEvent loggedEvent) {
+      if (actual.key() != -1) {
+        Assertions.assertThat(actual.key()).isEqualTo(loggedEvent.getKey());
+      }
+      final var loggedValue = ReflectUtil.newInstance(actual.recordValue().getClass());
+      loggedValue.wrap(
+          loggedEvent.getValueBuffer(), loggedEvent.getValueOffset(), loggedEvent.getValueLength());
+      final var loggedMetadata = new RecordMetadata();
+      loggedMetadata.wrap(
+          loggedEvent.getMetadata(),
+          loggedEvent.getMetadataOffset(),
+          loggedEvent.getMetadataLength());
+
+      Assertions.assertThat(actual.recordValue()).isEqualTo(loggedValue);
+      Assertions.assertThat(actual.recordMetadata()).isEqualTo(loggedMetadata);
+      return this;
+    }
+  }
+
+  public static class TestLogAppendEntryBuilder {
+    private long key = -1L;
+    private int sourceIndex = -1;
+    private UnifiedRecordValue recordValue = new TestRecordValue();
+    private RecordMetadata recordMetadata = new RecordMetadata().intent(Intent.UNKNOWN);
+
+    public TestLogAppendEntryBuilder withKey(final long key) {
+      this.key = key;
+      return this;
+    }
+
+    public TestLogAppendEntryBuilder withSourceIndex(final int sourceIndex) {
+      this.sourceIndex = sourceIndex;
+      return this;
+    }
+
+    public TestLogAppendEntryBuilder withRecordValue(final UnifiedRecordValue recordValue) {
+      this.recordValue = recordValue;
+      return this;
+    }
+
+    public TestLogAppendEntryBuilder withRecordMetadata(final RecordMetadata recordMetadata) {
+      this.recordMetadata = recordMetadata;
+      return this;
+    }
+
+    public LogAppendEntry build() {
+      return new TestEntry(key, sourceIndex, recordValue, recordMetadata);
+    }
+  }
+
+  public static final class TestRecordValue extends UnifiedRecordValue {}
```

Comment:
:wrench: I don't think this is required, `UnifiedRecordValue` can be instantiated directly.